code/datums/components/squeak.dm

```diff
@@ -71,25 +71,34 @@
 	else
 		steps++
 
-/datum/component/squeak/proc/play_squeak_crossed(atom/movable/AM)
-	if(isitem(AM))
-		var/obj/item/I = AM
+/datum/component/squeak/proc/play_squeak_crossed(atom/source, atom/movable/crossing)
+	if(istype(crossing, /obj/effect))
+		return
+	if(isitem(crossing))
+		var/obj/item/I = source
 		if(I.flags & ABSTRACT)
 			return
-		else if(isprojectile(AM))
-			var/obj/item/projectile/P = AM
-			if(P.original != parent)
+	if(isprojectile(crossing))
+		var/obj/item/projectile/P = crossing
+		if(P.original != parent)
+			return
+	if(ismob(source))
+		var/mob/M = source
+		if(M.flying)
+			return
+		if(isliving(source))
+			var/mob/living/L = M
+			if(L.floating)
 				return
-	if(ismob(AM))
-		var/mob/M = AM
+	if(ismob(crossing))
```

Comment:
Can you make these into an `else if` chain?

Reply:
Which level(s) do you want turned into elif?

Reply:
Just the `isprojectile`, `ismob` etc!